### PR TITLE
Tech/use context

### PR DESCRIPTION
**Pull request by iza-pik**
BEFORE CREATING THIS PR, have you bumped the package JSON where appropriate? Yes, v1.1.2

Ticket / Why this is needed: Rewriting List to use Context

Link to Figma doc:

### CODE REVIEW

File: src/Header/Header.stories.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { EdozoLogo2 } from '../EdozoLogo2';
 import { Header } from './Header';
 import { Menu } from '../Menu';
 import { Popover } from '../Popover';
-import { List, ListItem } from '../List';
+import { ListGroup } from '../List';
```
Comment (martinmcneela) [inline src/Header/Header.stories.tsx:9]:
```suggestion
import { List } from '../List';
```

Don't think we need the rename here

File: src/List/ListGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    </ListContext.Provider>
+  );
+};
+
+ListGroup.defaultProps = {
+  variant: 'menu',
+  disabled: false,
+};
+
+ListGroup.Item = ListItem;
```
Comment (martinmcneela) [inline src/List/ListGroup.tsx:35]: Like the use of compound components 👍

File: src/List/ListItem/ListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { useListContext } from '../ListGroup';
+import { StyledListItem, ListItemProps } from './ListItem.styles';
+
```
Comment (martinmcneela) [inline src/List/ListItem/ListItem.tsx:4]: Like how the item is in a nested folder here, make it easier to follow IMO 😃

File: src/List/ListGroup.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  </ListGroup>
+);
+
+export const SingleItemList: Story = args => (
+  <ListGroup {...args}>
+    <ListGroup.Item key={Math.random()}>
+      <ControlledListItem {...ControlledListItem.args} />
+    </ListGroup.Item>
+  </ListGroup>
+);
```
Comment (martinmcneela) [inline src/List/ListGroup.stories.tsx:57]: Think you can remove both of these stories, they are covered in the others

File: src/List/ListGroup.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      <DisabledListItem {...DisabledListItem.args} />
+    </ListGroup.Item>
+  </ListGroup>
+);
+
+export const SingleItemList: Story = args => (
+  <ListGroup {...args}>
+    <ListGroup.Item key={Math.random()}>
+      <ControlledListItem {...ControlledListItem.args} />
+    </ListGroup.Item>
```
Comment (martinmcneela) [inline src/List/ListGroup.stories.tsx:55]: These will be deleted but worth watching out for this, this nest a list item in a list item and results in the extra padding you have in these 2 stories

**Review by martinmcneela — CHANGES_REQUESTED** [review]

File: src/Header/Header.stories.tsx (hunk 1 above)
Comment (iza-pik) [reply src/Header/Header.stories.tsx]: That's true, I renamed it back to `List` now.

File: src/List/ListGroup.stories.tsx (hunk 5 above)
Comment (iza-pik) [reply src/List/ListGroup.stories.tsx]: Ah, yes, I had a separate template for these and missed this when I removed it.

File: src/List/List.styles.ts [hunk 6]
```diff
@@ -1,18 +1,7 @@
...
   background-color: ${p => p.theme.colors.white};
-  border-radius: ${p => p.theme.borderRadius[p.radius || 'standard']};
-  > div {
-    border-radius: ${p => p.theme.borderRadius[p.radius || 'standard']};
-  }
-
-  :not(:first-child) {
-    border-radius: 0;
-  }
+  border-radius: inherit;
```
Comment (iza-pik) [inline src/List/List.styles.ts:6]: Ah, this is so much better! 🙂

File: src/List/List.stories.tsx [hunk 7]
```diff
@@ -17,30 +14,31 @@ export default {
...
+    { node: 'Third', disabled: false },
+  ],
+};
 
-export const SingleItemList: Story = args => (
-  <List {...args}>
-    <ControlledListItem {...ControlledListItem.args} />
-  </List>
-);
+export const DisabledItemOnly: Story = () => <List.Item disabled>You can&#39;t click me</List.Item>;
```
[the remark this answers is not in the document]
Comment (martinmcneela) [reply src/List/List.stories.tsx:44]: you know, I don't think we need that story as it is highlighted in the List

File: src/List/ListItem/ListItem.styles.ts [hunk 8]
```diff
@@ -3,100 +3,46 @@ import styled, { css } from 'styled-components';
...
 
 export const StyledListItem = styled.div<ListItemProps>`
+  cursor: pointer;
   color: ${p => p.theme.colors.grayDark};
   font-weight: 700;
   background-color: ${p => p.theme.colors.white};
   position: relative;
   padding: ${p => p.theme.spacing.xsmall};
+  :not(:last-child) {
+    border-bottom: 1px solid ${p => p.theme.colors.gray};
```
Comment (iza-pik) [inline src/List/ListItem/ListItem.styles.ts:16]: Actually we've lost the opacity here on `border-bottom`, I added that previously to align with the design.

**Review by martinmcneela — APPROVED** [review]

**Pull request merged**